src/thor/bidirectional_astar.cc

```diff
@@ -240,15 +240,21 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
 
   // Get cost. Separate out transition cost.
   Cost transition_cost = costing_->TransitionCost(meta.edge, nodeinfo, pred);
-  Cost newcost = pred.cost() + transition_cost +
-                 costing_->EdgeCost(meta.edge, tile, kConstrainedFlowSecondOfDay);
+  Cost newcost =
+      pred.cost() + transition_cost +
+      costing_->EdgeCost(meta.edge, tile,
+                         kConstrainedFlowSecondOfDay + pred.cost().secs + transition_cost.secs);
```

Comment:
i remember you mentioning that you needed to add time tracking to bidirectional a* but this seems a bit fishy. can you drop a comment here about the intent.

the reason the constant is used it to force the default speed lookup behavior to favor day time driving speeds. the change here could (over a long enough route, 6 hours) push the time to default night speeds. im not sure that is desired here but i need a bit more time to look at what the constraint update in costing does

Reply:
Yes I understand why you default to noon.  The problem is that I need to track the time I arrive at an edge, because the hazard decays over time; before the change, every edge was initialized to noon, even if one arrived at that edge an hour later. Perhaps there's a better option?

Reply:
the reason this got my attention is im working on a branch where i will be tracking the procession of time over expansion (from both ends). ive added to that both the time of day but also how much time has passed. the next step of that PR will be to send this timing object into the edge cost function. where we can use the time of day to pick the speed of the edge (and there for the correct path). since it seems you need this too i think we s hould get that merged first and refactor this to use it instead of hijacking non-time dependent routes to become time dependent.

see the problem with the change above is that it has the potential to change path selection based on the time of day, which we explicitly didnt want to do for routes that were not time-dependent. i guess what you are saying is that for secondary constraints to be used you need time-dependence. but im guessing you also didnt want to bother with the timedependent algorithms since bidirectional is a more performant implementation.

let me know if i understood any of that incorrectly. i should get the time tracking done in the next week or so. sorry again for the wait!

Reply:
Yes, I'm using the bidirectional for the performance gain. The cost function for the hazard is somewhat expensive to calculate, so minimizing the search space is quite important. 

Waiting for the time tracking to get finished makes sense, then I will merge those changes and we can go from there.